katex.js

```diff
@@ -68,6 +70,25 @@ const generateParseTree = function(
     return parseTree(expression, settings);
 };
 
+/**
+ * If the given error is a KaTeX ParseError and options.throwOnError is false,
+ * renders the invalid LaTeX as a span with hover title giving the KaTeX
+ * error message.  Otherwise, simply throws the error.
+ */
+const renderError = function(
+    error,
+    expression: string,
+    options: Settings,
+) {
+    if (options.throwOnError || !(error instanceof ParseError)) {
+        throw error;
```

Comment:
Does re-throwing maintain the call stack?

Reply:
@kevinbarabash Yes 😄